### PR TITLE
Lu 4635

### PR DESCRIPTION
**Pull request by DarKim59**
Changes to encrypt password for Add user and Change password

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,4 @@
-FROM python:3.7-alpine
-
-RUN apk update && \
-    apk add gcc g++ && \
-    mkdir -p /deploy
+FROM python:3.7
```
Comment (salmaniqbal) [inline Dockerfile]: Why was the image changed? We don't want to change the image from python:3.7-alpine to python:3.7. This significantly increases the size of the base image from 98.4MB to 917MB. Takes longer to build and start up. Please change this back to 3.7-alpine and add any dependencies that you need.

File: blaise_survey_manager/users/functions.py [hunk 2]
```diff
@@ -35,5 +38,27 @@ def process_users_from_json(request):
                 db.session.commit()
 
 
+def encrypt_string(unencrypted_string):
+    log.info('encrypt_string start')
+
+    try:
+        log.info('unencrypted string : {}'.format(unencrypted_string))
```
Comment (salmaniqbal) [inline blaise_survey_manager/users/functions.py]: not good practice to log the unencrypted string or key. Or even the encrypted one. We want to reduce the number of places where the information can be got hold of.

Comment (salmaniqbal) [reply blaise_survey_manager/users/functions.py]: You can add general logging though, 'encrypting string' etc

File: bsm.yaml.tpl [hunk 3]
```diff
@@ -29,6 +29,11 @@ spec:
             path: /
             port: 5000
         env:
+          - name: PW_PUBKEY_TOKEN
+            valueFrom:
+              configMapKeyRef:
+                name: pubkey-pw-config
```
Comment (salmaniqbal) [inline bsm.yaml.tpl]: This needs to be added to terraform script as well. Speak with @cadmiumcat & @Riceyo to get it added there.

**Comment by salmaniqbal** [comment]
Make sure that thsi works in https://github.com/ONSdigital/Blaise_Developer_Services. Update the db alembic script to include encrypted password

File: Dockerfile (hunk 1 above)
Comment (DarKim59) [reply Dockerfile]: Now using Python 3.7-alpine again

File: blaise_survey_manager/users/functions.py (hunk 2 above)
Comment (DarKim59) [reply blaise_survey_manager/users/functions.py]: yes - removed logging of key

File: bsm.yaml.tpl (hunk 3 above)
Comment (DarKim59) [reply bsm.yaml.tpl]: Yes, I've given the details to Richmond

**Pull request merged**